cluster-autoscaler/cloudprovider/aws/api/instance_info.go

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	instanceInfoCacheMaxAge      = time.Hour * 6
+	awsPricingAPIURLTemplate     = "https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/%s/index.json"
+	instanceOperatingSystemLinux = "Linux"
+	instanceTenancyShared        = "Shared"
+)
+
+// InstanceInfo holds AWS EC2 instance information
+type InstanceInfo struct {
+	InstanceType  string
+	OnDemandPrice float64
+	VCPU          int64
+	MemoryMb      int64
+	GPU           int64
+}
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// NewEC2InstanceInfoService is the constructor of instanceInfoService which is a wrapper for AWS Pricing API.
+func NewEC2InstanceInfoService(client httpClient) *instanceInfoService {
+	return &instanceInfoService{
+		client: client,
+		cache:  make(instanceInfoCache),
+	}
+}
+
+type instanceInfoService struct {
+	client httpClient
+	cache  instanceInfoCache
+	mu     sync.RWMutex
```

Comment:
don't use single/two letter names for field in struct. 
moreover you can do 
type xxxx struct {
     sync.RwMutex // <<<NO FIELD NAME.
}
and then just xxx.Lock().